internal/entitlement/httpdriver/entitlement.go

```diff
@@ -328,6 +329,10 @@ func (h *entitlementHandler) ListEntitlements() ListEntitlementsHandler {
 					return entitlement.EntitlementType(s)
 				}),
 			}
+			if !p.Page.IsZero() {
```

Comment:
I don't know how many other places we have the same optional structure, but we can consider creating a helper that takes a pageNumber pointer and pageSize pointer and constructs the page from that.

Reply:
We could do a helper for it, the reason this is separate is because we also have to support limit & offset

We could do a helper method that takes the limit, offset, page, pageSize pointers and constructs the output....

Reply:
In practice we should never reimplement this, its only there to support the legacy limit/offset pagination as well